### PR TITLE
AD 150 - Part 1

### LINKED ISSUE #252: Cultures to add and revise
(Waiting for the CK3 new patch to drop before working on it)

- [x] Galatian (from modded I:R _galatian_) / issue #239 
- [x] Hunnic (no I:R source)
- [x] Proto-Baltic and Proto-Slavic (from I:R _aestian_)
- [x] Various Iron Age Italian cultures: Oscan from _osco_umbrian_ and _umbrian_, Samnite from _samnite_, Venetic from _venetic_, Ligustic from _ligurian_, Messapic from _lucanian_ and _bruttian_, Sardonian from _sardonian_, Siculian from _siculian_
- [x] Adnanite and Qahtanite, to represent pre-Islamic Arabs.

Mappings to correct:
- [x] Wusun should be mapped to Pecheneg instead of Sogdian. Pecheneg-Wusun identity is a historical hypothesis, but surely more reliable than just shoving them into Sogdian
- [x] Shule to Saka. Quoth Wikipedia, _Much like the neighboring people of the Kingdom of Khotan, people of [the Shule Kingdom] spoke Saka, one of the Eastern Iranian languages._
- [x] Yuezhi to Bactrian. According to the spoken language, which is not Greek.

From forum:
- [ ] Mede 1-1 mapping? [https://forum.paradoxplaza.com/forum/threads/imperator-to-ck3-release-thread.1415172/post-27769127](https://forum.paradoxplaza.com/forum/threads/imperator-to-ck3-release-thread.1415172/post-27769127) - not implemented; the Mede-Kurd mapping is convenient because it gives us a source for CK3 Kurdish culture and doesn't require adding a Median culture to blankMod
- [x] Elamite 1-1 mapping? [https://forum.paradoxplaza.com/forum/threads/imperator-to-ck3-release-thread.1415172/post-27769127](https://forum.paradoxplaza.com/forum/threads/imperator-to-ck3-release-thread.1415172/post-27769127)
- [x] Magadhi to Bengali [https://forum.paradoxplaza.com/forum/threads/imperator-to-ck3-release-thread.1415172/post-27763343](https://forum.paradoxplaza.com/forum/threads/imperator-to-ck3-release-thread.1415172/post-27763343)

### LINKED ISSUE #254: Improve text for culture_era_late_classical_desc

### PR DESCRIPTION
**Pull request by IohannesIohannium**
- Resolves #254
- Resolves #252 (and #239)
- Solves parts of #249
- No CTD found on char's death

### CODE REVIEW

**Comment by IohannesIohannium** [comment]
Crash source found: something is not working correctly with eras, so I've restored the vanilla setup for it

File: ImperatorToCK3/Data_Files/blankMod/output/common/culture/eras/~00_culture_eras.txt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#culture_era_tribal = {
+#	year = 535 # Start of Gothic War
+#	modifier = { # Restoring base CK3 stuff
+#		tyranny_gain_mult = -0.25
+#		army_maintenance_mult = 0.5
+#		diplomatic_range_mult = 0.25
+#		title_creation_cost_mult = -100
+#		movement_speed = 0.25
+#	}
+#}
```
Comment (IhateTrains) [inline ImperatorToCK3/Data_Files/blankMod/output/common/culture/eras/~00_culture_eras.txt:30]: I guess you need to check which of these modifiers caused the crash. Are that many modifiers needed in the first place?

Comment (IohannesIohannium) [reply ImperatorToCK3/Data_Files/blankMod/output/common/culture/eras/~00_culture_eras.txt:30]: Yes and no, this was just a provisional revision of eras to fit seven more centuries of game. As I will get tech modding, I will see and fix this.

**Comment by IohannesIohannium** [comment]
Updated files after #305

**Comment by IhateTrains** [comment]
Does it still crash after 3 years?

File: ImperatorToCK3/Data_Files/configurables/culture_map.txt [hunk 2]
```diff
@@ -197,7 +185,7 @@ link = { ck3 = saxon imp = saxonian imp = anglian imp = gothonic imp = raumarici
...
+link = { ck3 = frisian region = custom_netherlands imp = saxonian imp = anglian imp = gothonic imp = raumarician imp = suiones imp = reudingian imp = cimbrian imp = rugian imp = herulian imp = ingvaeonic imp = istvaeonic imp = teutonian imp = gutones imp = lemovian imp = bastarnae imp = irminonic imp = helveconian imp = suebian imp = frisian imp = vandal ck3Province = 2115 ck3Province = 2125 ck3Province = 2105 ck3Province = 2106 ck3Province = 2406 ck3Province = 2405 ck3Province = 2112 ck3Province = 2450 ck3Province = 2428 ck3Province = 2429 ck3Province = 2424 ck3Province = 2421 ck3Province = 2417 ck3Province = 2414 ck3Province = 2418 ck3Province = 2413 }
```
Comment (IhateTrains) [inline ImperatorToCK3/Data_Files/configurables/culture_map.txt:188]: `region` is not a valid keyword, replace it with `ck3Region`

File: ImperatorToCK3/Data_Files/configurables/culture_map.txt [hunk 3]
```diff
@@ -246,7 +234,7 @@ link = { ck3 = thracian imp = thracian }
...
 
 ## Illyrian
-link = { ck3 = albanian impRegion = macedonia_region impRegion = greece_region impRegion = magna_graecia_region ck3Province = 470 ck3Province = 3717 ck3Province = 3720 ck3Province = 3718 ck3Province = 3719 ck3Province = 3721 ck3Province = 3724 ck3Province = 3723 ck3Province = 3722 ck3Province = 3713 ck3Province = 3714 ck3Province = 3715 ck3Province = 3716 ck3Province = 3520 ck3Province = 3711 ck3Province = 3712 imp = taulantian imp = dardanian imp = messapian imp = deurian imp = deraemestian imp = catari imp = liburnian_culture imp = histrian imp = abrian imp = greco_illyrian }
+link = { ck3 = albanian region = macedonia_region region = greece_region region = magna_graecia_region ck3Province = 470 ck3Province = 3717 ck3Province = 3720 ck3Province = 3718 ck3Province = 3719 ck3Province = 3721 ck3Province = 3724 ck3Province = 3723 ck3Province = 3722 ck3Province = 3713 ck3Province = 3714 ck3Province = 3715 ck3Province = 3716 ck3Province = 3520 ck3Province = 3711 ck3Province = 3712 imp = taulantian imp = dardanian imp = messapian imp = deurian imp = deraemestian imp = catari imp = liburnian_culture imp = histrian imp = abrian imp = greco_illyrian }
```
Comment (IhateTrains) [inline ImperatorToCK3/Data_Files/configurables/culture_map.txt:237]: Replace region with impRegion, this is already fixed in master

**Review by IhateTrains — CHANGES_REQUESTED** [review]
Some reverted fixes in culture map, rest seems ok

File: ImperatorToCK3/Data_Files/configurables/culture_map.txt (hunk 2 above)
Comment (IohannesIohannium) [reply ImperatorToCK3/Data_Files/configurables/culture_map.txt]: Done

File: ImperatorToCK3/Data_Files/configurables/culture_map.txt (hunk 3 above)
Comment (IohannesIohannium) [reply ImperatorToCK3/Data_Files/configurables/culture_map.txt]: Done

**Review by IhateTrains — APPROVED** [review]

**Pull request merged**